forum/backend.py

```diff
@@ -14,6 +14,11 @@ def is_mysql_backend_enabled(course_id: str | None) -> bool:
         from forum.toggles import ENABLE_MYSQL_BACKEND  # pylint: disable=C0415
     except ImportError:
         return True
+    # pylint: disable=C0415,E0401
+    from opaque_keys.edx.locator import CourseKey  # type: ignore[import-not-found]
+
+    if isinstance(course_id, str):
+        course_id = CourseKey.from_string(course_id)
```

Comment:
can we move opaquekey import to try block or simply remove try-except?

Reply:
Done, moved the import to the try-except block.